pallets/pips/src/lib.rs

```diff
@@ -1302,4 +1298,8 @@ impl<T: Trait> Module<T> {
         };
         proposal_data
     }
+
+    pub fn is_proposal_id_valid(id: PipId) -> bool {
+        <Proposals<T>>::contains_key(id)
```

Comment:
I think we should also check that the `id` is valid in terms of being enacted / rejected. The logic for this would be:
```
pips::is_referendum_state(id, ReferendumState::Pending)?;
```